xdsl/backend/csl/print_csl.py

```diff
@@ -133,17 +141,51 @@ def print_block(self, body: Block):
                     self.print(
                         f"const {self._get_variable_name_for(r)} : {type_name} = {value_str};"
                     )
-                case func.FuncOp(sym_name=name, body=bdy, function_type=ftyp) if len(
+                case csl.ImportModuleConstOp(module=module, params=params, result=res):
+                    name = self._get_variable_name_for(res)
+
+                    params_str = ""
+                    if params is not None:
+                        params_str = f", {self._get_variable_name_for(params)}"
+
+                    self.print(
+                        f'const {name} : comptime_struct = @import_module("{module.data}"{params_str});'
```

Comment:
Giving the result of `@import_module` an explicit type of `comptime_struct` crashes the cslc compiler. The type can be left out or explicitly specified as `imported_module`. AFAICT this is the only difference between`comptime_struct` and `imported_module`, so no other changes would be necessary (except in the filecheck).

Reply:
Good to know! I added a new type `ImportedModuleType` and an alias `StructLike = ImportedModuleType | ComptimeStructType`, which we can extend with any other struct like objects in the future, I feel this gives us the best balance between accuracy and flexibility without lying in the IR types (compared to when we just print a different return type from an import)